### ISSUE TITLE
Indicate to Global Caches how long a resource should be cached for

### ISSUE THREAD

**Issue by 6a6d74** [wmo-im/wis2-guide#43]
As discussed at ET-W2AT meeting, 16-Oct-2023.

WIS2 Manual (Tech Regs) states:

"""
4.5.4 Based on received notifications, a Global Cache shall download core data from WIS nodes or other Global Caches and store for a duration compatible with the real-time or near real-time schedule of the data and not less than 24-hours.
"""

The default period for caching resources is 24-hours. But some (near) real time might need to be kept for longer based on the "real-time or near real-time schedule". Its a WMO Programme decision about how long data should be cached for, agreed with the WIS2 operators - in a process that's yet to be defined.

A mechanism is required to inform Global Caches to cache content for a duration other than 24-hours - usually longer, but perhaps shorter too.

It seems likely that this would be a per-object policy, i.e., each object that is cached should indicate the length of time it should be cached. The simplest way to do this would be to add a `cache-duration` attribute to the notification message.

A Global Cache may ignore the `cache-duration` statement, for example, if the duration was set to an excessively long period that compromised the ability of the Global Cache to perform. In "tech regs speak": the Global Cache **should** respect the `cache-duration` value.

**Comment by amilan17** [comment]
note: current guide differs from text above content. https://wmo-im.github.io/wis2-guide/guide/wis2-guide-DRAFT.html#_2_4_3_global_cache

**Comment by amilan17** [comment]
Do we need to amend the WIS2 Guide?

**Comment by golfvert** [comment]
Maybe, but not now, for me.

As a principle, in the coming 2/3 years (while WIS2.0 is still in its infancy) we should change manual/guide on WIS2 **ONLY** to fix what is not working. 
The `cache-duration` might be a good idea.  Right now, we have an agreed mechanism for GC not to cache data if they don't want to. They have a wait to do it. How to inform the centres. Let's stick to that.

Politically, having Fast Tracked changes as we have for BUFR, GRIB,... is to be avoided as much as possible in WIS2.0 specs.

At least, this is what I think :) We can discuss this general approach in an upcoming W2IT meeting.

**Comment by efucile** [comment]
I agree with @golfvert on keeping the changes on WIS2 Manual and Guide to minimum.  Keeping up with fast track changes is a challenge for everyone.
However, this might be something broken. In the sense that if the cache implementation is focused on the minimum 24 hours some data may have a permanence too short for their use.
I think it's worth a discussion at W2IT meeting.

**Comment by golfvert** [comment]
In addition to my initial comment, I don't think we should engage on this road...
If someone asks for a cache of eg 3-days, and the GC cache only (for whatever reason) for 1-day.
What do we do ? GC publishes something on deletion ?
I am afraid we may open a can of worms to address, IMHO, not a critical problem.

**Comment by 6a6d74** [comment]
There's no compelling reason at this time to add this functionality to WIS2. It is _potentially_ a good idea. Let's review after INFCOM4. Putting this issue on the parking lot.
